src/iter/mod.rs

```diff
@@ -1384,6 +1387,14 @@ pub trait ParallelIterator: Sized + Send {
         sum::sum(self)
     }
 
+    fn scan<F>(self, scan_op: F, id: Self::Item) -> Scan<Self::Item, F>
```

Comment:
Let's start here, documenting at the API level, especially for a user who is looking for an equivalent to `Iterator::scan`. How would you describe what this does, and importantly how is it different than the sequential version? The signature of `F` is quite different, looking more like some flavor of `reduce`.

Reply:
Good question, I wasn't exactly sure what the signature should be, and it could change. But it's similar to the way `reduce` compares to `fold`. This should have an identical result to sequential scan when the operation doesn't have internal state, and is associative. If not, it won't make sense to run it in parallel.

Reply:
Hi @cuviper, could you take another look at this?

Reply:
I don't feel you addressed my first comment? The signature is part of it, but please make an attempt at adding documentation describing what this does. And frankly, I don't think many people even use `Iterator::scan`, so it's not sufficient to just reference that. Suppose this existed independently - describe what it does, and what points are important for the user to think about.

Reply:
Sorry, I misunderstood you last time. I've added documentation.

I was hoping to get some feedback on the signature and general approach. These are the differences from the other ParallelIterator functions, and my rationale for them:
1. The scan operation currently has type `Fn(&Item, &Item) -> Item`, since when we call it iteratively, we need to keep each intermediate result and not consume it. 
2.  Since `scan_op` takes in references, we don't need multiple copies of `identity`. So, I have `identity` as type `Item`, rather than `Fn() -> Item`, since it's simpler. The other functions use `Fn() -> Item`, though, and it would be easy to change. 

Do you have any preferences on those?